task_sdk/src/airflow/sdk/api/client.py

```diff
@@ -123,6 +125,16 @@ def heartbeat(self, id: uuid.UUID, pid: int):
         body = TIHeartbeatInfo(pid=pid, hostname=get_hostname())
         self.client.put(f"task-instances/{id}/heartbeat", content=body.model_dump_json())
 
+    def defer(self, id: uuid.UUID, msg):
+        body = DeferTask(
+            state="deferred",
+            classpath=msg.classpath,
+            trigger_kwargs=msg.trigger_kwargs,
+            next_method=msg.next_method,
+            trigger_timeout=msg.timeout,
+        )
```

Comment:
Isn't msg a `DeferTask` object anyway?

https://github.com/apache/airflow/blob/73cf2acd8f06ab96e2917083ff3652c147fab415/task_sdk/src/airflow/sdk/execution_time/supervisor.py#L531-L533

Reply:
Good catch. I just refactored this part in the next commit